sdk/src/main/java/io/radar/sdk/RadarBeaconManager.kt

```diff
@@ -88,16 +88,18 @@ internal class RadarBeaconManager(
             return
         }
 
-        this.addCallback(callback)
+        if (beacons.isEmpty()) {
+            logger.d(this.context, "No beacons to range")
 
-        if (this.started) {
-            logger.d(this.context, "Already ranging beacons")
+            callback?.onComplete(RadarStatus.SUCCESS)
 
             return
         }
 
-        if (beacons.isEmpty()) {
-            logger.d(this.context, "No beacons to range")
+        this.addCallback(callback)
+
+        if (this.started) {
+            logger.d(this.context, "Already ranging beacons")
```

Comment:
for future consideration: `beacons` could be a different set of beacons than the ones already being scanned. Just mentioning to create awareness of the current behavior.

Reply:
Good callout